lib/bamboo/attachment.ex

```diff
@@ -2,26 +2,27 @@ defmodule Bamboo.Attachment do
   @moduledoc """
   """
 
-  defstruct filename: nil, content_type: nil, path: nil
+  defstruct filename: nil, content_type: nil, path: nil, data: nil
 
   @doc ~S"""
   Creates a new Attachment
 
   Examples:
-    Attachment.new("/path/to/attachment.png")
-    Attachment.new("/path/to/attachment.png", filename: "image.png")
-    Attachment.new("/path/to/attachment.png", filename: "image.png", content_type: "image/png")
-    Attachment.new(params["file"]) # Where params["file"] is a %Plug.Upload
+    Attachment.new_from_file("/path/to/attachment.png")
```

Comment:
Also I think `new` is better here since there isn't a second function. Alternatively you could add `new_from_data` so there were two different functions. I'd probably prefer just `new` though